pkgs/applications/networking/instant-messengers/element/element-desktop.nix

```diff
@@ -104,6 +104,7 @@ mkYarnPackage rec {
     extraEntries = ''
       StartupWMClass=element
       MimeType=x-scheme-handler/element;
+      X-macOS-Squircle=true
```

Comment:
I though about it, but Apple’s algorithm only seems to work well when the icon has a solid background. Of the icons I tried (Element, FFXIV, kitty, and File Roller), Element was the only one that looked okay.

FFXIV got mangled, and kitty lost half the icon (because the squircle background is always black regardless of the icon). File Roller was technically fine, but macOS shrunk it down instead of cropping it. I think it would have looked better and closer to the original design if it were cropped.

Given the potential for problems, I erred on the side of preserving upstream’s icons over trying to force them into the macOS 11+ style. If it turns out most apps are actually fine with doing that, and these are actually rare cases, the default can be changed.

Actually, if we did that, it might be worth looking at doing the work ourselves, so we can have control over the output (background color, crop versus shrink, etc). That might not be a bad idea for a future change, but it’s probably beyond the scope of this PR.

Reply:
I spent some time playing with this a bit more. I was wrong. There is no logic to how macOS creates the squircle. It just masks the underlying icon with the squircle mask. I just happened to find icons that looked bad when that was done. For something like kitty’s icon, you’d probably want to composite it with a background color. Otherwise, you always get black. When I push my other change, I’ll go ahead and make the squircle the default.